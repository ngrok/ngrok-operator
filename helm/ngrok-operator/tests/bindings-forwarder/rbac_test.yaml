suite: test bindings forwarder rbac
templates:
- bindings-forwarder/rbac.yaml
set:
  bindings.enabled: true
tests:
- it: Should match snapshot
  asserts:
  - matchSnapshot: {}
- it: should not create a role or role binding if bindings.enabled is false
  set:
    bindings.enabled: false
  asserts:
  - hasDocuments:
      count: 0
- it: should create a role if bindings.enabled is true
  documentIndex: 0
  asserts:
  - isKind:
      of: Role
  - isAPIVersion:
      of: rbac.authorization.k8s.io/v1
  - equal:
      path: metadata.name
      value: RELEASE-NAME-ngrok-operator-bindings-forwarder
- it: should create a rolebinding if bindings.enabled is true
  documentIndex: 1
  asserts:
  - isKind:
      of: RoleBinding
  - isAPIVersion:
      of: rbac.authorization.k8s.io/v1
  - equal:
      path: metadata.name
      value: RELEASE-NAME-ngrok-operator-bindings-forwarder-rolebinding
