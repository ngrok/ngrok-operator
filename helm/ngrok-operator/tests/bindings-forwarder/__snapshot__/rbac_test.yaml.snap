Should match snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      name: RELEASE-NAME-ngrok-operator-bindings-forwarder
      namespace: NAMESPACE
    rules:
      - apiGroups:
          - bindings.k8s.ngrok.com
        resources:
          - boundendpoints
        verbs:
          - get
          - list
          - watch
          - patch
          - update
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - create
          - patch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: RELEASE-NAME-ngrok-operator-bindings-forwarder-rolebinding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: RELEASE-NAME-ngrok-operator-bindings-forwarder
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-ngrok-operator-bindings-forwarder
        namespace: NAMESPACE
